tests/utils.go

```diff
@@ -2235,6 +2235,15 @@ func NewRandomFedoraVMIWitGuestAgent() *v1.VirtualMachineInstance {
 	)
 }
 
+func NewRandomFedoraVMIWitGuestAgentNetworkData(networkData string) *v1.VirtualMachineInstance {
```

Comment:
Instead of introducing this method, I would create the default network data inside the `NewRandomFedoraVMIWitGuestAgent` method.